tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/op/core/Init.java

```diff
@@ -89,18 +89,20 @@ public static Init create(Scope scope) {
    *
    * <p>Registered initializers are then grouped as a single unit of computation by adding
    * and executing an {@link org.tensorflow.op.core.Init#create(Scope) init} operation from a graph
-   * session.
+   * session.  No-ops if the session is eager.
    *
    * @param scope
    * @param initializer
-   * @throws IllegalArgumentException if the execution environment in scope is not a graph
    * @see org.tensorflow.op.core.Init#create(Scope) init
    */
   @Endpoint(name = "initAdd")
   public static void add(Scope scope, Op initializer) {
     ExecutionEnvironment exEnv = scope.env();
+    if(exEnv.isEager()){
+      return;
+    }
     if (!(exEnv instanceof Graph)) {
-      throw new IllegalArgumentException("initAdd is only supported on Graph sessions.");
+      throw new IllegalArgumentException("initAdd is only supported on Graph or eager sessions.");
```

Comment:
I think pretty much all over the code, while we "might" have to support later a third execution environment type, we kind of assume that there is only to. So I would only simplify this method with:
```java
if (exEnv.isGraph()) {
   Graph graph = (Graph) exEnv;
   graph.addInitializer(initializer);
}
```